sse2neon.h

```diff
@@ -112,18 +113,60 @@
 /* If using MSVC */
```

Comment:
My original change had this check:

```
#if defined(__cplusplus)
#else
#error sse2neon only supports C++ compilation with this compiler
#endif
```

It'd still be good to have this message so users aren't hit with random compiler errors when compiling as C with MSVC.